### PR TITLE
Use in-memory thumbnail APIs when possible

### PR DESCRIPTION
**Pull request by jmartinesp**
## Content

- Instead of using `Client.getMediaFIle(...)` to load thumbnails, use `Client.getMediaThumbnail(...)`. This fixes a bug with upcoming new Rust APIs where thumbnails for videos weren't being displayed.
- Add thumbnail dimensions info to `TimelineItemImageContent` and `TimelineItemVideoContent`.
- For images in timeline items, try loading the thumbnail source if it exists, use the actual media source otherwise.
- Add default values for thumbnail sizes if none is provided.
- Removes redundant `ByteArray.toUByteArray().toByteArray()` transformation. This was needed before because uniffi returned something like `Array<UByte>` instead, but it now returns a proper `ByteArray`.

## Motivation and context

Fix issue in upcoming SDK versions, use the right APIs when possible.

## Tests

- Clear the cache.
- Open a room with media items.

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 14

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemImageView.kt [hunk 1]
```diff
@@ -86,13 +88,23 @@ fun TimelineItemImageView(
...
+                        MediaRequestData(
+                            source = content.thumbnailSource,
+                            kind = MediaRequestData.Kind.Thumbnail(
+                                width = content.thumbnailWidth?.toLong() ?: MAX_THUMBNAIL_WIDTH,
+                                height = content.thumbnailHeight?.toLong() ?: MAX_THUMBNAIL_HEIGHT,
+                            ),
+                        )
+                    } else {
+                        MediaRequestData(
+                            source = content.thumbnailSource,
```
Comment (ganfra) [inline features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemImageView.kt:101]: I think we should keep `content.preferredMediaSource` and always try to use `kind = MediaRequestData.Kind.Thumbnail`

Comment (ganfra) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemImageView.kt:101]: Also, the thumbnailAPI is probably not working with GIF?

Comment (jmartinesp) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemImageView.kt:101]: Nope, it doesn't work properly... I guess we'll need to check the mime type here too.

Comment (ganfra) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemImageView.kt:101]: This is already done in `preferredMediaSource`, so maybe we could have a `mediaRequestData` extension function in `Content`, to pick the correct `source` and `kind`?

Comment (jmartinesp) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemImageView.kt]: I tried something like that in https://github.com/element-hq/element-x-android/pull/3817/commits/6da362e499428bbc3fd68eaf8bce57f21d8e3a2c.

Comment (ganfra) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemImageView.kt]: But why do you want to use `Kind.File` when using `mediaSource` (except for isMimeTypeAnimatedImage I mean).
I'd have done something like that : 

```
val thumbnailMediaRequest: MediaRequestData by lazy {
        if (mimeType.isMimeTypeAnimatedImage()) {
            MediaRequestData(
                source = mediaSource,
                kind = MediaRequestData.Kind.File(
                    fileName = filename,
                    mimeType = mimeType
                )
            )
        } else {
            MediaRequestData(
                source = thumbnailSource ?: mediaSource,
                kind = MediaRequestData.Kind.Thumbnail(
                    width = thumbnailWidth?.toLong() ?: MAX_THUMBNAIL_WIDTH,
                    height = thumbnailHeight?.toLong() ?: MAX_THUMBNAIL_HEIGHT
                ),
            )
        }
    }
```

and removed the `preferredMediaSource`?

Comment (jmartinesp) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemImageView.kt]: Used this in https://github.com/element-hq/element-x-android/pull/3817/commits/daba49cac0b560f105ae8257595f457a3cb57e59.

**Comment by jmartinesp** [comment]
Just a heads up: this is needed to fix a bug with the media upload + send queue implementation, so it should be merged soon-ish.

**Review by ganfra — APPROVED** [review]
Thanks for the changes, LGTM!

**Pull request merged**